packages/cli/src/commands/init/init.ts

```diff
@@ -534,12 +571,24 @@ export default (async function initialize(
 
   validateProjectName(projectName);
 
-  if (!!options.template && !!options.version) {
+  const version = await npmResolveConcreteVersion(
+    'react-native',
+    options.version ?? DEFAULT_VERSION,
+  );
+
+  // From 0.75 it actually is useful to be able to specify both the template and react-native version.
+  // This should only be used by people who know what they're doing.
+  if (semver.gte(version, TEMPLATE_COMMUNITY_REACT_NATIVE_VERSION)) {
+    logger.warn(
+      `Use ${chalk.bold('--template')} and ${chalk.bold(
+        '--version',
+      )} only if you know what you're doing. Here be dragons 🐉.`,
+    );
```

Comment:
How is this going to work if a user specifies:

```
template @react-native-community/template@0.75.1 --version 0.75.2
```

?

I think we should fail here instead

Reply:
It'll fail for all version before `0.75.*`, but I think this is a valid workflow that people testing unreleased versions (release team + developers on CLI) would find useful.  We add a warning which is pretty clear.

I think we should keep it, but we can add a hidden argument (something like `--ci` or `--know-better`) to allow the power users I'm targeting to use this combination.